core/app/models/spree/payment_create.rb

```diff
@@ -24,13 +32,14 @@ def initialize(order, attributes, payment: nil, request_env: {})
     # Build the new Payment
     # @return [Payment] a new (unpersisted) Payment
     def build
+      @payment ||= order.payments.find(attributes[:id]) if attributes[:id]
       @payment ||= order.payments.new
       @payment.request_env = @request_env if @request_env
       @payment.attributes = @attributes
 
       if source_attributes[:wallet_payment_source_id].present?
         build_from_wallet_payment_source
-      else
+      elsif source_attributes.present?
```

Comment:
Thoughts on using a guard clause instead?

Reply:
I gave it a try but wasn't really happy with the result, and preferred to have that if/elsif as the central body of the method, but thanks for suggesting it 👍